dist/web3-light.js

```diff
@@ -3327,6 +3348,145 @@ module.exports = ICAP;
     You should have received a copy of the GNU Lesser General Public License
     along with ethereum.js.  If not, see <http://www.gnu.org/licenses/>.
 */
+/** @file ipcprovider.js
+ * @authors:
+ *   Fabian Vogelsteller <fabian@ethdev.com>
+ * @date 2015
+ */
+
+"use strict";
+
+var utils = require('../utils/utils');
+var errors = require('./errors');
+
+
+var IpcProvider = function (path, net) {
+    var _this = this;
+    this.responseCallbacks = {};
+    this.path = path;
+    
+    net = net || require('net');
+
+
+    try {
+        this.connection = net.connect({path: this.path});
+    } catch(error) {
+        throw errors.InvalidConnection(path);
+    }
+
+    // this.connection.on('error', function(e){
+    //     throw errors.InvalidConnection(path);
+    // }); 
+
+
+
+    // LISTEN FOR CONNECTION RESPONSES
+    this.connection.on('data', function(result) {
+        result = result.toString();
+
+        try {
+            result = JSON.parse(result);
+
+        } catch(e) {
+            throw errors.InvalidResponse(result);                
+        }
+
+        var id;
+
+        // get the id which matches the returned id
+        if(utils.isArray(result)) {
+            result.forEach(function(load){
+                if(_this.responseCallbacks[load.id])
+                    id = load.id;
+            });
+        } else {
+            id = result.id;
+        }
+
+
+        // fire the callback
+        if(_this.responseCallbacks[id]) {
+            _this.responseCallbacks[id](null, result);
+            delete _this.responseCallbacks[id];
+        }
+
+    });
+};
+
+/**
+Get the adds a callback to the responseCallbacks object,
+which will be called if a response matching the response Id will arrive.
+
+@method _getResponse
+*/
+IpcProvider.prototype._getResponse = function(payload, callback) {
+    var id = payload.id || payload[0].id;
+
+    this.responseCallbacks[id] = callback;
+};
+
+
+IpcProvider.prototype.isConnected = function() {
+    // try reconnect, when connection is gone
+    if(!this.connection._handle)
+        this.connection.connect({path: this.path});
+
+    return !!this.connection._handle;
+};
+
+IpcProvider.prototype.send = function (payload) {
+
+    if(this.connection.writeSync) {
```

Comment:
Its not supported by the native `net` package, but part of the mist net wrapper.

The mist wrapper is able to do synchronous IPC connections to the electron backend (which then sends of the real write) and therefore we can still provide synchronous send in mist. When you use it in node directly we would throw an error stating that sync is not supported as IPC connections in node a inherently async.